crypto/s2n_hash.c

```diff
@@ -24,7 +24,7 @@
 
 int s2n_hash_hmac_alg(s2n_hash_algorithm hash_alg, s2n_hmac_algorithm *out)
 {
-    PRECONDITION_POSIX(S2N_MEM_IS_READABLE(out, sizeof(*out)));
+    ENSURE_POSIX(S2N_MEM_IS_READABLE(out, sizeof(*out)), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
Should there be a macro for this?

Reply:
Probably a good idea. Although it should probably be a separate PR since this one is getting big :)

Reply:
I'm tracking your request with the issue https://github.com/awslabs/s2n/issues/2461.